packages/flutter_tools/test/general.shard/build_system/targets/asset_transformer_test.dart

```diff
@@ -0,0 +1,377 @@
+    final BufferLogger logger = BufferLogger.test();
+    final Artifacts artifacts = Artifacts.test();
+
+    final File asset = fileSystem.file('asset.txt')..createSync();
+    const String outputPath = 'output.txt';
+
+    final String dartBinaryPath = artifacts.getArtifactPath(Artifact.engineDartBinary);
+    final FakeProcessManager processManager =
+        FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: <Pattern>[
+          dartBinaryPath,
+          'run',
+          'my_transformer',
+          RegExp('--input=.*'),
+          RegExp('--output=.*'),
+        ],
+        onRun: (_) {
+          // Do nothing.
+        },
+        stderr: 'Transformation failed, but I forgot to exit with a non-zero code.'
+      ),
+    ]);
+
+    final AssetTransformer transformer = AssetTransformer(
+      processManager: processManager,
+      logger: logger,
+      fileSystem: fileSystem,
+      dartBinaryPath: dartBinaryPath,
+    );
+
+    await transformer.transformAsset(
+      asset: asset,
+      outputPath: outputPath,
+      workingDirectory: fileSystem.currentDirectory.path,
+      transformerEntries: <AssetTransformerEntry>[
+        const AssetTransformerEntry(
+          package: 'my_transformer',
+          args: <String>[],
+        )
+      ],
+    );
+
+    expect(processManager, hasNoRemainingExpectations);
+    expect(logger.errorText, contains(RegExp(
+r'''
+User-defined transformation of asset "asset\.txt" failed.
+Asset transformer my_transformer did not produce an output file\.
+Input file provided to transformer: ".*\.txt"
+Expected output file at: ".*\.txt"
+Full command: Artifact\.engineDartBinary run my_transformer --input=.*\.txt --output=.*\.txt
+stdout:
+
+stderr:
+Transformation failed, but I forgot to exit with a non-zero code\.
+'''
+    )));
+  });
+
+  testWithoutContext('correctly chains transformations when there are multiple of them', () async {
+    final FileSystem fileSystem = MemoryFileSystem();
+    final BufferLogger logger = BufferLogger.test();
+    final Artifacts artifacts = Artifacts.test();
+
+    final File asset = fileSystem.file('asset.txt')
+      ..createSync()
+      ..writeAsStringSync('ABC');
+    const String outputPath = 'output.txt';
+
+    final String dartBinaryPath = artifacts.getArtifactPath(Artifact.engineDartBinary);
+    final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: <Pattern>[
+          dartBinaryPath,
+          'run',
+          'my_lowercase_transformer',
+          RegExp('--input=.*'),
+          RegExp('--output=.*'),
+        ],
+        onRun: (List<String> args) {
+          final ArgResults parsedArgs = (ArgParser()
+              ..addOption('input')
+              ..addOption('output'))
+            .parse(args);
+
+          final String inputFileContents = fileSystem.file(parsedArgs['input']).readAsStringSync();
+          fileSystem.file(parsedArgs['output'])
```

Comment:
let's check the contents equal `'ABC'`

Reply:
Why?

Reply:
Isn't this implicitly already tested from the output file's contents being a function of the input file's contents?

Reply:
good point, SGTM